x-pack/plugins/fleet/server/services/agent_policy.ts

```diff
@@ -869,6 +875,9 @@ class AgentPolicyService {
       await pMap(
         agentPolicies,
         async (agentPolicy) => {
+          const soClient = appContextService.getInternalUserSOClientForSpaceId(
+            agentPolicy.space_ids?.[0]
```

Comment:
does the logic work correctly if there are more than one `space_ids`?

Reply:
Yes need a client that can access the policy so any of the space from `space_ids` will work